app/decorators/typed_model.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  ColumnOptions,
+  LucidModel,
+  ModelColumnOptions,
+} from "@adonisjs/lucid/types/model";
+
+type ColumnType = "string" | "number" | "boolean" | "DateTime";
+
+type TypedModelOptions = Record<string, ColumnType>;
+
+export interface ColumnDef extends ColumnOptions {
+  meta?: {
+    type?: string;
+  };
+}
+
+/**
+ * Class decorator for injecting custom column types into a Lucid model.
+ *
+ * The `typedModel` decorator allows you to specify and inject custom types for model columns by
+ * adding type information to the `meta` property of each column definition.
+ *
+ * **Prerequisites:**
+ * - The model must extend `BaseModel` from AdonisJS Lucid ORM.
+ * - Columns should be defined using column decorators (e.g., `@column`).
+ *
+ * @example
+ * ```typescript
+ * import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'
+ * import { typedModel } from 'path-to-decorator'
+ *
+ * @typedModel({
+ *   id: 'number',
+ *   name: 'string',
+ *   createdAt: 'date',
+ * })
+ * export default class User extends BaseModel {
+ *   @column({ isPrimary: true })
+ *   public id: number
+ *
+ *   @column()
+ *   public name: string
+ *
+ *   @column.dateTime({ autoCreate: true })
+ *   public createdAt: DateTime
+ * }
+ * ```
+ *
+ * @param {TypedModelOptions} options - An object where:
+ *   - **Keys**: Match the column names defined in the model.
+ *   - **Values**: Are types from the predefined `ColumnType` set (e.g., `'string'`, `'number'`, `'boolean'`, `'DateTime'`).
+ *
+ */
+export function typedModel<T extends LucidModel>(options: TypedModelOptions) {
+  return function (constructor: T): T {
+    const columns: Map<string, ModelColumnOptions> | undefined =
+      constructor.$columnsDefinitions;
+
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    if (columns === undefined) {
```

Comment:
question(linter): @Rei-x can you take a look at this line?

Reply:
My guess is that because that field in the LucidModel is defined as just a Map, with no possibility of an undefined value, eslint can see that `columns` will never actually be undefined, even though it's defined here as possibly undefined.
![image](https://github.com/user-attachments/assets/c8f9fd72-ff58-4055-ac46-10b7fdcfabfe)